src/odemis/driver/orsay.py

```diff
@@ -1389,3 +1440,1247 @@ def terminate(self):
+                new_translation[1] += 0.5  # prefer adding a pixel to the top
+        new_translation = tuple(new_translation)
+
+        self.imageFormatUpdatedResolutionTranslation.wait(60)  # wait until the image format has updated image area
+        # This is needed, because changing the image format on the server, sets the translation to (0.0, 0.0) and
+        # resolution equal to the new image format. We don't want that. We want to keep the resolution and translation
+        # as they were (except for appropriate scaling).
+
+        self.resolution.value = new_resolution  # set new resolution with calling the setter
+        self.translation.value = new_translation  # set new translation with calling the setter
+
+        logging.debug("Updating imageFormat to %s and updating translation and resolution and their ranges accordingly."
+                      % str(value))
+        return value
+
+    def _updateImageFormat(self, parameter=None, attributeName="Actual"):
+        """
+        Reads the image format from the Orsay server and saves it in the imageFormat VA
+
+        :param (Orsay Parameter) parameter: The parameter on the Orsay server that calls this callback
+        :param (str) attributeName: The name of the attribute of parameter which was changed
+        """
+        if parameter is None:
+            parameter = self._ionColumn.ImageSize
+        if parameter is not self._ionColumn.ImageSize:
+            raise ValueError("Incorrect parameter passed to _updateImageFormat. Parameter should be "
+                             "datamodel.IonColumnMCS.ImageSize. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName != "Actual":
+            return
+        state = self._ionColumn.ImageSize.Actual
+        logging.debug("Image format is: %s. Updating translation and resolution and their ranges accordingly." % state)
+        new_value = tuple(map(int, state.split(" ")))
+        self.imageFormat._value = new_value  # to not call the setter
+        self.imageFormat.notify(new_value)
+
+    def _translation_setter(self, value):
+        """
+        Setter of the translation VA.
+
+        :param (tuple (float, float)) value: Target translation of the area to image
+        :return (tuple (float, float)): The actual translation set
+
+        The translation VA marks the centre of the image area with respect to the centre of the field of view. This
+        setter transforms the coordinates of the centre of the image area to the coordinates of the top left corner of
+        the image area, which is the format the Orsay server takes. The setter also adjusts the size of the image area
+        (resolution VA) to prevent the new translation from placing part of the image area outside of the image format.
+        """
+        self.updatingImageArea.acquire()  # translation and resolution cannot be updated simultaniously
```

Comment:
This raises a big flag "troubles ahead": you set explicitly a lock, and only release it into another function.
Firstly, if any line in this function fails, the lock will be there forever. Secondly, if the other function is never called, the lock will also stay there forever. For instance, is this function called if by coincidence we request the current value?

I think that one possibility would be to use `with self.updatingImageArea` around the VA setter functions. They would then wait for an event (imageFormatUpdatedResolutionTranslation?), which the parameter update function would send when it's completed.

I'm not completely sure it'd work out of the box though. Maybe it'd be simpler/safer to have a single function which is called to change the image area/size. Then this function would take care of blocking until the request has been applied (or a timeout happens).